packages/playwright-core/src/client/locator.ts

```diff
@@ -328,7 +328,11 @@ export class Locator implements api.Locator {
   waitFor(options: channels.FrameWaitForSelectorOptions & { state: 'attached' | 'visible' }): Promise<void>;
   waitFor(options?: channels.FrameWaitForSelectorOptions): Promise<void>;
   async waitFor(options?: channels.FrameWaitForSelectorOptions): Promise<void> {
-    await this._frame._channel.waitForSelector({ selector: this._selector, strict: true, omitReturnValue: true, ...options });
+    return this._frame._wrapApiCall<void>(async () => {
+      const result = await this._frame._channel.waitForSelector({ selector: this._selector, strict: true, omitReturnValue: true, ...options });
```

Comment:
I believe omitReturnValue was supposed to avoid handle allocation but got lost during refactorings. Let's either remove 'omitReturnValue' and all dispose here as you do or make 'omitReturnValue' work.